cmd/registry/cmd/diff/diff.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2020 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package diff
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/apigee/registry/cmd/registry/core"
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/log"
+	"github.com/apigee/registry/rpc"
+	"github.com/apigee/registry/server/registry/names"
+	"github.com/hexops/gotextdiff"
+	"github.com/hexops/gotextdiff/myers"
+	"github.com/hexops/gotextdiff/span"
+	"github.com/spf13/cobra"
+	"google.golang.org/api/iterator"
+)
+
+func Command(ctx context.Context) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "diff",
+		Short: "Compare resources in the registry",
+		Args:  cobra.MinimumNArgs(2),
+		Run: func(cmd *cobra.Command, args []string) {
+			client, err := connection.NewClient(ctx)
+			if err != nil {
+				log.FromContext(ctx).WithError(err).Fatal("Failed to get client")
+			}
+
+			var spec1, spec2 *rpc.ApiSpec
+			var path1 string
```

Comment:
Consider making this a `names.Spec` instead. I don't see any real benefit to making it a string prior to the moment it's being concatenated in the other function. Without the type it wasn't clear to me what "path" actually was until I looked at everything that set it.

Reply:
OK, if you think so - updated as suggested.